js/goto.js

```diff
@@ -1,27 +1,42 @@
+import {
+  SIGNUP,
+  JOIN,
+  HOME,
+  SIGNUP_URL,
+  JOIN_URL,
+  HOME_URL,
+} from './constants.js';
+
+const hasJoinVisited = localStorage.getItem('hasJoinVisited');
+
 function redirectUserToPage(page) {
   const finalPage =
-    page == 'signup'
-      ? 'https://my.realdevsquad.com/signup?state=get-started'
-      : 'https://realdevsquad.com';
+    page == SIGNUP ? SIGNUP_URL : page == JOIN ? JOIN_URL : HOME_URL;
 
   window.location.href = finalPage;
 }
 
+function redirectionHandler(data) {
+  if (data.incompleteUserDetails) {
+    redirectUserToPage(SIGNUP);
+  } else if (hasJoinVisited == 'true' || hasJoinVisited == null) {
```

Comment:
If I am not wrong, condition has to be 
```js
else if(hasJoinVisited !== 'true' || hasJoinVisited === null)
{}
```

Reply:
If user go through the join page url then `hasJoinVisited == true` or user go through any site(Except join page url) then `hasJoinVisited == null`  then we redirected to join page section. After completed the join section `hasJoinVisited == false` then user redirect to home page.